src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -923,7 +955,11 @@ private static void ParseHeaderNameValue(Span<byte> line, HttpResponseMessage re
 
             // Note we ignore the return value from TryAddWithoutValidation; 
             // if the header can't be added, we silently drop it.
-            if (descriptor.HeaderType == HttpHeaderType.Content)
+            if (isFromTrailer)
+            {
+                response.TrailingHeaders.TryAddWithoutValidation(descriptor.HeaderType == HttpHeaderType.Request ? descriptor.AsCustomHeader() : descriptor, headerValue);
```

Comment:
Nit: can you move the comment below about request headers to be above the if so that it applies to both this if branch and the final else branch?

Reply:
(It would also be good to include in the comment why they must be treated as custom, as it's not obvious now why that's the case.  @geoffkizer should be able to answer.)

Reply:
Yes, I would be curious to know as well.